### DISCUSSION TITLE
How the redis db has been used, and what are the touch points in codebase?

### DISCUSSION THREAD

**Discussion by patelb2012** [chatwoot/chatwoot#3820] [Q&A]
Hi there, I was exploring the self-hosted Chatwoot and reading the [documentation](https://www.chatwoot.com/docs/self-hosted) for same. But did not found the details where and how the redis db has been used. Can anyone share correct link/information where I can find the details.

**Comment by sojan-official** [comment]
Chatwoot uses Redis to keep a cache of specific information which requires frequent read / writes. 
Examples: online status, agent round-robin queue etc

You can see the full list of keys over here : 
https://github.com/chatwoot/chatwoot/blob/develop/lib/redis/redis_keys.rb



logged https://github.com/chatwoot/docs/issues/201 for the team to add detailed documentation
